packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -33,10 +35,22 @@ export const useSignUpContext = (): SignUpContextType => {
   const { componentName, ...ctx } = (React.useContext(ComponentContext) || {}) as SignUpCtx;
   const { navigate } = useRouter();
   const { displayConfig } = useEnvironment();
-  const { queryParams } = useRouter();
+  const { queryParams, queryString } = useRouter();
   const options = useOptions();
   const clerk = useCoreClerk();
 
+  const initialValuesFromQueryParams = useMemo(() => {
```

Comment:
🔧 We can make this drier and pass the `SIGN_(UP|IN)_INITIAL_VALUE_KEYS' as a parameter.

Reply:
I created a function and reused it.